### ISSUE TITLE
Red Faction 2 - crash on exit

### ISSUE THREAD

**Issue by WinterSnowfall** [AlpyneDreams/d8vk#126]
The game stopped behaving nicely on exit at some point, although it used to work just fine. WineD3D still behaves nicely on exit, so I don't think it's a Wine problem. I'll have to bisect it some day, but I'm raising this issue so that we don't forget it's a regression.

I've captured a trace but I couldn't reproduce the crash while capturing, so it's sort of useless.

### Software information
Red Faction 2

### System information
- GPU: RTX 3060
- Driver: 525.85.05
- Wine version: 8.4 Staging
- DXVK version: 2c56a1a

### Apitrace file(s)
[rf2.trace.tar.xz](https://drive.google.com/file/d/1vePMWDPro6VcBK6OlgYISOQrMc6loOyc/view?usp=share_link)

### Log files
- d3d8.log: [rf2_d3d8.log](https://github.com/AlpyneDreams/d8vk/files/11097534/rf2_d3d8.log)

**Comment by WinterSnowfall** [comment]
I haven't done an actual bisect yet, but can confirm the regression predates the 0.10 release. So, note to self (or anyone else who wants to beat me to it), the bisect range should be between 0585064 and 03ff0bb.

**Comment by WinterSnowfall** [comment]
```
faff2d700cda52641d0c54b09ecc34bed7fb25f2 is the first bad commit
commit faff2d700cda52641d0c54b09ecc34bed7fb25f2
Author: Alpyne <alpyne.dreams@gmail.com>
Date:   Sat Nov 5 23:58:17 2022 -0800

    [d3d8] Cleanup texture and surface

 src/d3d8/d3d8_surface.h |  9 ---------
 src/d3d8/d3d8_texture.h | 21 ---------------------
 2 files changed, 30 deletions(-)

```
I'll see if I can figure out what the game doesn't like from the above.

**Comment by WinterSnowfall** [comment]
\*Sigh\*. Apparently this is not a regression, but an older problem resurfacing. The game has always had problems with crashing on exit, even with the commits I considered problem-free, only they used to be far less frequent. The addition of subresource caching (1652611) somehow made them more prevalent, but there are still times when the game exits just fine even with latest main.

I can't reproduce the issue at all with WineD3D, so this is definitely something related to d8vk/dxvk.

**Comment by WinterSnowfall** [comment]
Fixed by bd394b8 :tada: .
